packages/serve/src/lib/middleware/docServerMiddleware.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  BuiltInMiddleware,
+  DocumentServer,
+  KoaContext,
+  Next,
+} from '@vulcan-sql/serve/models';
+import {
+  VulcanInternalExtension,
+  TYPES as CORE_TYPES,
+  DocumentOptions,
+} from '@vulcan-sql/core';
+import { inject, interfaces } from 'inversify';
+import { TYPES } from '@vulcan-sql/serve/containers';
+import * as compose from 'koa-compose';
+
+@VulcanInternalExtension()
```

Comment:
How about to setup the all different document server type under one options module name, because seems different document server extensions both use the same `DocumentOptions`.

like below:

```yaml
api-doc:
  spec: [oas3]
  server: [redoc, ...]
```

And your document server middleware read the `doc` options, and according to your `server` to create each document type server ?

Reply:
Did you mean to change to property name from `document` to `api-doc` and turn it from internal options to module configuration?

I think it will be great for renaming, but because it a cross-package config, that is, spec for both `build` and `serve` package and server for `serve` package, I'd prefer to use internal options to do some validations.